funnel/templates/event.html

```diff
@@ -46,37 +46,251 @@
+              //Participant has been checked-in so remove from 'checkin-queue' 
+              window.Queue.dequeue("checkin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+          else {
+            if($(participant).hasClass('js-checkin')) {
+              //Participant's check-in has been cancelled so remove from 'cancelcheckin-queue' 
+              window.Queue.dequeue("cancelcheckin-queue", participantID);
+            }
+            else {
+              $(participant).siblings('.loader').show();
+            }
+          }
+        });
+      }
+    }
+
+    function getParticipantsData() {
+      $.ajax({
+        type: 'GET',
+        url:  '{{ "{0}event/{1}/participants/json".format(space.url_for(), event.name) }}',
+        timeout: 5000,
+        dataType: 'jsonp',
+        success: function(data) {
+          participantTableContent.set('participants', data.participants).then(function() {
+            if (Modernizr.localstorage) {
+              $('.loader').hide();
+              updateQueue("checkin-queue");
+              updateQueue("cancelcheckin-queue");
+            }
+          });
+        }
+      });
+    }
+
+    function postCheckinStatus(participantID, action) {
+      if(action) {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID + "?checkin=t";
+      }
+      else {
+        var posturl = "{{ '{0}event/{1}/checkin/'.format(space.url_for(), event.name) }}" + participantID;
+      }
+      $.ajax({
+        type: 'post',
+        url:  posturl,
+        timeout: 5000,
+        dataType: "json",
+        success: function(data) {
+          if(data.checked_in) {
+            window.Queue.dequeue("checkin-queue", data.participant_id);
+          }
+          else {
+            window.Queue.dequeue("cancelcheckin-queue", data.participant_id);
+          }
+        }
+      });
+    }
+
+    function checkQueues() {
+      //Read first participant from 'checkin-queue' and post check-in to server
+      var participantID = window.Queue.peek("checkin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, true);
+      }
+      //Read first participant from 'cancelcheckin-queue' and post cancel check-in to server
+      participantID = window.Queue.peek("cancelcheckin-queue");
+      if(participantID) {
+        postCheckinStatus(participantID, false);
+      }
+    }
+
+    //Get the list of participants from server
+    getParticipantsData();
+
+    //If localStorage is available, register event handlers for check-in, cancel-checkin and abort
+    if (Modernizr.localstorage) {
+      $('#participants-table-content').on('click', '.js-checkin', function(event) {
+        event.preventDefault();
+        //Add to participant_id to "checkin-queue"
+        window.Queue.enqueue("checkin-queue", $(this).data('pid'));
```

Comment:
> Instance of Queue means handling another copy of check-in and cancel queue.

Why is this a problem? Also, let's consider:

````javascript
    var firstItem = window.Queue.first("cancelcheckin-queue");
    window.Queue.dequeue("cancelcheckin-queue", data.participant_id);
````
versus

````javascript
   var firstItem = cancelCheckinQ.first();
   cancelCheckinQ.dequeue(data.participant_id);
````
As a reader, I find the latter approach more readable, and easier to understand. I have a clear idea about when the queue is being instantiated. Also, I stand to get hints from a [linter](http://jshint.com/) if I try to use a queue that is uninitialized or out of scope. Finally, I don't have to specify the name of localStorage key, every time I need to perform an operation. These points become especially useful when managing multiple queues.

An analogy is the data models on the back-end. For instance:
````python
    class Proposal(BaseScopedIdNameMixin, CoordinatesMixin, db.Model):
        __tablename__ = 'proposal'
       ...
````
The `tablename` is well encapsulated and I only have to worry about the `Proposal` and the instances I create, instead of having to know the name of the table for every time I would need to perform operations on the model.